helix-term/src/commands/typed.rs

```diff
@@ -1095,13 +1095,18 @@ fn change_current_directory(
             .editor
             .last_cwd
             .clone()
-            .ok_or(anyhow!("No previous working directory"))?,
+            .ok_or_else(|| anyhow!("No previous working directory"))?,
         Some(input_path) => helix_stdx::path::expand_tilde(Path::new(input_path)).to_path_buf(),
         None => home_dir()?,
     };
 
     cx.editor.last_cwd = helix_stdx::env::set_current_working_dir(dir)?;
 
+    let (_, doc) = current!(cx.editor);
+
+    // Clear the relative path as this will need to be recalculated.
+    doc.relative_path.take();
```

Comment:
All relative paths will need to be reset so this will need to go through `cx.editor.documents_mut()`